src/auth/src/hash.rs

```diff
@@ -79,6 +80,14 @@ pub fn hash_password(password: &Password) -> Result<PasswordHash, HashError> {
     })
 }
 
+pub fn generate_nonce(client_nonce: &str) -> String {
+    let mut nonce = [0u8; 24];
+    openssl::rand::rand_bytes(&mut nonce).unwrap();
```

Comment:
Not sure where we do a catch_unwind but may want to but may want to return an error instead?